mongo/pagination/pagination.go

```diff
@@ -0,0 +1,172 @@
+package pagination
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/Scalingo/go-utils/mongo/document"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type BadRequestError struct {
+	Errors map[string][]string `json:"errors"`
+}
+
+func (err BadRequestError) Error() string {
+	errArray := make([]string, 0, len(err.Errors))
+	for errTitle, errValues := range err.Errors {
+		errArray = append(errArray, fmt.Sprintf("* %s → %s", errTitle, strings.Join(errValues, ", ")))
+	}
+	return strings.Join(errArray, "\n")
+}
+
+func NewBadRequestErrors() *BadRequestError {
+	return &BadRequestError{
+		Errors: make(map[string][]string),
+	}
+}
+
+func NewPaginationService(opts ServiceOpts) Service {
+	return opts
+}
+
+type Meta struct {
+	CurrentPage int  `json:"current_page"`
+	PrevPage    *int `json:"prev_page"`
+	NextPage    *int `json:"next_page"`
+	TotalPages  int  `json:"total_pages"`
+	TotalCount  int  `json:"total_count"`
+	perPageNum  int
+}
+
+type Service interface {
+	Paginate(ctx context.Context, DBQuery bson.M, collection string, result interface{}, sortField string) (*Meta, error)
```

Comment:
```suggestion
	Paginate(ctx context.Context, DBQuery bson.M, collection string, result interface{}, sortField string) (Meta, error)
```
Shouldn't we return the struct instead of a pointer to it ?